contrib/mpi-proxy-split/lower-half/gethostbyname-static/gethostbyname_static.c

```diff
@@ -166,12 +177,7 @@ int getaddrinfo(const char *__restrict node,
     close(pipefdin[0]);
     dup2(pipefdout[1], 1);
     close(pipefdout[1]);
-    char *argv[] = {"getaddrinfo", (char *)node, (char *)service, NULL};
-    // argv[2] (hints) is passed in from parent via writeall
-    char buf[10000];
-    char *path = getProxyPath(buf, sizeof(buf));
-    int rc = execvp(path, argv);
-    if (rc == -1) { perror("execvp"); }
+    execvpProxyWithArgv("getaddrinfo", service);
```

Comment:
`node` is not passed in when it should be. i'd suggest changing the implementation to this:
```
char *argv[] = { (char *) node, (char *) service, NULL };
execvpProxyWithArgv("getaddrinfo", *argv);
```

Reply:
Thanks, @fyshhh.  This is a really important catch.
And then as I looked more closely, I realized that I was also not capturing the `ai_canonname` (canonical hostname) correctly either.  I've now fixed both in the new update..